api_reflector/views.py

```diff
@@ -105,18 +106,31 @@ def mock(path: str) -> Response:
     else:
         req_json = {}
 
-    templateable_request = rules_engine.TemplatableRequest(
-        params=params, json=req_json, query=request.args, headers=request.headers
-    )
-    response = rules_engine.find_best_response(templateable_request, response_rules)
-
-    content = template_env.from_string(response.content).render(
-        {
-            "request": templateable_request,
-            **default_context,
-        }
-    )
-
-    response.execute_actions(req_json, content)
+    try:
+        templateable_request = rules_engine.TemplatableRequest(
+            params=params, json=req_json, query=request.args, headers=request.headers
+        )
 
-    return Response(content, status=response.status_code, mimetype=response.content_type)
+        response = rules_engine.find_best_response(templateable_request, response_rules)
+        try:
+            content = template_env.from_string(response.content).render(
+                {
+                    "request": templateable_request,
+                    **default_context,
+                }
+            )
+
+            response.execute_actions(req_json, content)
+        except Exception as ex:  # pylint: disable=broad-except
+            if isinstance(ex, UndefinedError):
+                msg = "Undefined field used in the response template"
+            elif isinstance(ex, TemplateSyntaxError):
+                msg = "Syntax error in response template"
```

Comment:
i think these two are the only kinds of exceptions we'd like to catch here, at least for now. we could also consider a fallback on the base `TemplateError` just to cover off all possible jinja failures.

rather than having the nested try blocks in here catching everything, it might make sense to do something like this instead:

```python
templateable_request = rules_engine.TemplatableRequest(
    params=params, json=req_json, query=request.args, headers=request.headers
)

try:
    response = rules_engine.find_best_response(templateable_request, response_rules)
    content = template_env.from_string(response.content).render(
        {
            "request": templateable_request,
            **default_context,
        }
    )
except UndefinedError as ex:
    ...
except TemplateSyntaxError as ex:
    ...
except TemplateError as ex:
    ...

return Response(content, status=response.status_code, mimetype=response.content_type)
```

this has a few advantages:

* indentation is reduced
* only code that can fail with template errors is included in the try block
* non-jinja errors are allowed to propagate (they are probably bugs to be fixed)

you may also wish to define a function that lets you return an error with a custom message in the three `except` blocks to reduce repetition.

regarding the status code, i'm not entirely sure what we should go with. i don't think 406 is correct as MDN says it indicates issues with the content type negotiation - i.e. the client has passed an `Accept` header that the server can't fulfull.

i think we should consider a 5xx status code, since the client's request was fine but the server is unable to respond. (RFC wording: "The server failed to fulfill an apparently valid request.")

we could make up our own, or we could just go with a generic 500 and be done with it. i don't think it matters too much which one we go with.

let me know what you think!

Reply:
Thanks for reviewing.

I had the same thought about avoiding the nested try-except. But the reason for doing that was to a clear separation in a `Templatable request error` and `Response template error`. Hence the two status codes. 

But i think i agree. If we don't care about catching all the possible excepts for response. 

Decide to put in 406 because it suggests a server failing to produce a response rather than a client side error. But yeah its very much nothing to do with content-type

I think 500 but with a decently detailed response message is good enough?

Reply:
yeah, with this particular issue i'm only really concerned with catching template errors so that the user can understand that they've made a mistake in setting up their response. other types of errors, if they come up, can be handled in a separate issue.

i'm happy with a 500 status code & description. :)

Reply:
I've just updated, lmk what you think. Will squash the commits once ready to merge